### PR TITLE
Fix a linking error when building the unittest

### PR DESCRIPTION
**Pull request by Gyuyoung**
When building unittest target by using
"npm run test -- unit_tests -- Debug", there is
a build error as below,

```
ld.lld: error: undefined symbol: BraveTestingProfile::BraveTestingProfile()
>>> referenced by push_messaging_service_unittest.cc
>>>               obj/chrome/test/unit_tests/push_messaging_service_unittest.o:(PushMessagingServiceTest::PushMessagingServiceTest())
clang: error: linker command failed with exit code 1 (use -v to see invocation)
ninja: build stopped: subcommand failed.
null
null
```

This PR updates chrome-test-BUILD.gn.patch in order to add
brave_browser_tests_deps dependency to unit_tests target.

Resolves brave/brave-browser#10910.

## Submitter Checklist:

- [ ] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [ ] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [ ] Android
  - [ ] iOS
  - [x] Linux
  - [ ] macOS
  - [ ] Windows
- Verified that these changes pass automated tests (unit, browser, security tests) on
  - [ ] iOS
  - [ ] Linux
  - [ ] macOS
  - [ ] Windows
- [x] Verified that all lint errors/warnings are resolved (`npm run lint`)
- [ ] Ran `git rebase master` (if needed).
- [ ] Ran `git rebase -i` to squash commits (if needed).
- [ ] Tagged reviewers and labelled the pull request as needed.
- [ ] Request a security/privacy review as needed.
- [ ] Add appropriate QA labels (QA/Yes or QA/No) to include the closed issue in milestone
- [ ] Public documentation has been updated as necessary. For instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-compatibility-issues-with-tracking-protection
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions 
- [ ] Verify test plan is specified in PR before merging to source

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on.
- [ ] All relevant documentation has been updated.

### CODE REVIEW

**Comment by Gyuyoung** [comment]
CI reported an error related to brave:audit_deps. It looks like it's not related to this PR. WDYT?

```
22:35:07  > ninja -C /Users/jenkins/jenkins/workspace/pr-brave-browser-gyuyoung-build-fix-unittest/src/out/Component_audit brave:audit_deps
22:35:07  ninja: Entering directory `/Users/jenkins/jenkins/workspace/pr-brave-browser-gyuyoung-build-fix-unittest/src/out/Component_audit'
22:35:10  ninja: error: unknown target 'brave:audit_deps'
```

**Comment by Gyuyoung** [comment]
@jonathansampson I wonder if you can also take a look at this PR.

File: patches/chrome-test-BUILD.gn.patch [hunk 1]
```diff
@@ -10,3 +10,11 @@ index fbbea351bd8ad009c81a41a7d1ba0590611c21bb..edc96d7f0b52ae06b336ff6d32ca055d
      # Runtime dependencies
      data_deps = [
        "//chrome:browser_tests_pak",
+@@ -3699,6 +3700,7 @@ test("unit_tests") {
+   deps = [
+     ":test_support",
+     ":test_support_unit",
++    "//brave/test:brave_test_support_unit",
```
Comment (bridiver) [inline patches/chrome-test-BUILD.gn.patch:17]: we don't need this patch as mentioned in DM, but patches should always be extensible so if we did patch, we would do something like `deps += brave_chrome_unit_tests_deps`

Comment (Gyuyoung) [reply patches/chrome-test-BUILD.gn.patch:17]: I thought that "npm run test -- unit_tests -- Debug" built the Chromium unit_tests. But,  if unit_tests means brave_unit_tests or Brave doesn't maintain the unit_tests, this PR is not unnecessary. Thanks for reviewing.

Comment (bridiver) [reply patches/chrome-test-BUILD.gn.patch:17]: you should be running `brave_unit_tests` which includes `push_messaging_service_unittest.cc` in the sources

Comment (Gyuyoung) [reply patches/chrome-test-BUILD.gn.patch:17]: Ok.

**Pull request closed without merging**